dbt/adapters/athena/python_submissions.py

```diff
@@ -0,0 +1,208 @@
+            if session_status in ["FAILED", "TERMINATED", "DEGRADED"]:
+                raise DbtRuntimeError(f"The session chosen was not available. Got status: {session_status}")
+            if session_status == "IDLE":
+                break
+            time.sleep(polling_interval)
+            polling_interval *= 2
+            if polling_interval > self.timeout:
+                raise DbtRuntimeError(f"Session {self.session_id} did not become free within {self.timeout} seconds.")
+
+    def submit(self, compiled_code: str) -> Any:
+        """
+        Submit a calculation to Athena.
+
+        This function submits a calculation to Athena for execution using the provided compiled code.
+        It starts a calculation execution with the current session ID and the compiled code as the code block.
+        The function then polls until the calculation execution is completed, and retrieves the result.
+        If the execution is successful and completed, the result S3 URI is returned. Otherwise, a DbtRuntimeError
+        is raised with the execution status.
+
+        Args:
+            compiled_code (str): The compiled code to submit for execution.
+
+        Returns:
+            dict: The result S3 URI if the execution is successful and completed.
+
+        Raises:
+            DbtRuntimeError: If the execution ends in a state other than "COMPLETED".
+
+        """
+        while True:
+            try:
+                calculation_execution_id = self.athena_client.start_calculation_execution(
+                    SessionId=self.session_id, CodeBlock=compiled_code.lstrip()
+                )["CalculationExecutionId"]
+                break
+            except botocore.exceptions.ClientError as ce:
+                LOGGER.exception(f"Encountered client error: {ce}")
+                if (
+                    ce.response["Error"]["Code"] == "InvalidRequestException"
+                    and "Session is in the BUSY state; needs to be IDLE to accept Calculations."
+                    in ce.response["Error"]["Message"]
+                ):
+                    LOGGER.exception("Going to poll until session is IDLE")
+                    self.poll_until_session_idle()
+            except Exception as e:
+                raise DbtRuntimeError(f"Unable to complete python execution. Got: {e}")
+        execution_status = self.poll_until_execution_completion(calculation_execution_id)
+        LOGGER.debug(f"Received execution status {execution_status}")
+        if execution_status == "COMPLETED":
+            try:
+                result = self.athena_client.get_calculation_execution(CalculationExecutionId=calculation_execution_id)[
+                    "Result"
+                ]
+            except Exception as e:
+                LOGGER.error(f"Unable to poll execution status: Got: {e}")
+                result = {}
+        self.spark_connection.release_session_lock(self.session_id)
```

Comment:
Hi @Avinash-1394 

Just wanted to check if this is intentional that we won't terminate the session that is currently executing and will wait until the `execution_status == "COMPLETED"` even if the timeout is exceeded? 

In which case, in the function: `release_session_lock()` would we even come across a case where session will be in BUSY state while we terminate it (for timeout reason) ?

Reply:
That is a very good point! If we have exceeded the timeout it should raise that the model failed and release the session. 

> In which case, in the function: release_session_lock() would we even come across a case where session will be in BUSY state while we terminate it (for timeout reason) ?

That is a good question. Hopefully we dont encounter that scenario but as a failsafe maybe it can check if its busy and stop the calculation.

Reply:
I believe this was addressed in the PR that you merged

Reply:
Yes, all my comments on this PR are addressed. I didn't find an option to resolve the outdated conversations. Can you please resolve all my conversations if you see an option for that.

Thank you